Document/0x05h-Testing-Platform-Interaction.md

```diff
@@ -8,18 +8,64 @@ Android assigns a distinct system identity (Linux user ID and group ID) to every
+- **Contacts provider usage stats change**: when an app requests the [`READ_CONTACTS`](https://developer.android.com/reference/android/Manifest.permission.html#READ_CONTACTS "READ_CONTACTS") permission, queries for contact's usage data will return approximations rather than exact values (the auto-complete API is not affected by this change).
+
+Apps targeting Android 8, i.e. [only API level 26 or higher](https://developer.android.com/about/versions/oreo/android-8.0-changes#o-apps "Apps targeting Android 8.0") are affected by the following:
+
+- **Account access and discoverability improvements**: Apps can no longer get access to user accounts only by having the [`GET_ACCOUNTS`](https://developer.android.com/reference/android/Manifest.permission.html#GET_ACCOUNTS "GET_ACCOUNTS") permission granted, unless the authenticator owns the accounts or the user grants that access.
+- **New telephony permissions**: the following permissions (classified as dangerous) are now part of the `PHONE` permissions group:
+  - The `ANSWER_PHONE_CALLS` permission allows to answer incoming phone calls programmatically (via `acceptRingingCall`).
+  - The `READ_PHONE_NUMBERS` permission grants read access to the phone numbers stored in the device.
+- **Direct document access**: the new [`getDocumentUri`](https://developer.android.com/reference/android/provider/MediaStore.html#getDocumentUri(android.content.Context,%20android.net.Uri) "getDocumentUri") method can be used to locate media files. However, it doesn't grant apps permission to access those files.
+- **Wi-Fi scanning restrictions**: Calling `WifiManager.getScanResults` requires at least one of the following permissions: `ACCESS_FINE_LOCATION`, `ACCESS_COARSE_LOCATION`, `CHANGE_WIFI_STATE`.
+- **Restrictions when granting dangerous permissions**: Dangerous permissions are classified into permission groups (e.g. the `STORAGE` group contains `READ_EXTERNAL_STORAGE` and `WRITE_EXTERNAL_STORAGE`). Before Android 8, it was sufficient to request one permission of the group in order to get all permissions of that group also granted at the same time. This has changed [starting at Android 8](https://developer.android.com/about/versions/oreo/android-8.0-changes#rmp "Android 8 Permissions Changes"): whenever an app requests a permission at runtime, the system will grant exclusively that specific permission. However, note that **all subsequent requests for permissions in that permission group will be automatically granted** without showing the permissions dialog to the user. See this example from the Android developer documentation:
```

Comment:
As we are focusing on security, I would suggest to get rid of the permissions listed above and only keep the "- **Restrictions when granting dangerous permissions**:", as this is a security topic a developer and also pentester should be aware of. The rest shouldn't be mentioned in the MSTG and should be referenced in a link as I am afraid it becomes otherwise too detailed in the MSTG. For example:

With Android 8 (API Level 26) several [changes](https://developer.android.com/about/versions/oreo/android-8.0-changes#atap "Changes for all apps") were made that affect all apps running on Android 8, even to those apps targeting API levels lower than 26. The most significant from a security point of view is about restrictions when granting dangerous permissions...

Reply:
Thanks for the comment! I agree on that and have addressed this on the new changes. However, I'd like to keep the discussion on this for a bit: the question now is if we should leave all changes to permissions that directly affect access to sensitive data and/or user privacy. 

For the moment, I've left the changes related to READ_CONTACTS, GET_ACCOUNTS and the new dangerous PHONE permissions. If we are testing permissions in relation to access to sensitive data, then it might be helpful to have such an overview. Please let me know what you think?

Reply:
I would say whatever is relevant to security we should keep in, permissions with access to sensitive data would definitely be one of that. 
Reads better now, thanks!